netket/experimental/driver/tdvp.py

```diff
@@ -295,17 +301,27 @@ def run(
         callbacks = _to_iterable(callback)
         callback_stop = False
 
-        with tqdm(total=self.t + T, disable=not show_progress) as pbar:
-            old_step = self.step_value
-            first_step = True
+        with tqdm(
+            total=self.t + T,
+            disable=not show_progress,
+            unit_scale=True,
+        ) as pbar:
+            self._first_step = True
 
+            self._pbar = pbar
```

Comment:
This is very, very brittle...

Reply:
It needs to be provided to `self.iter` somehow and adding a private argument is ugly, IMHO.

(I just noticed a bug because I forgot to add an `if self._pbar is not None` to `iter`, so I cannot really argue against this being brittle though.)

Reply:
I noticed that we do not need to add any argument to `iter`, just add a private method `_iter` which accepts the additional callback and call that from `iter` and `run`. I agree this is much cleaner now.